app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -136,6 +138,31 @@ export default class AMRHeatmapView extends React.Component {
     }
   };
 
+  onGeneLabelClick = geneName => {
+    const { sidebarVisible, sidebarMode, selectedGene } = this.state;
+    if (!geneName) {
+      this.setState({
+        sidebarVisible: false,
+      });
+      return;
+    }
+    if (
```

Comment:
it looks like you could just combine these two `ifs` like `(!geneName || sidebarVisible && ...)`